src/test/java/com/contrast/labs/ai/mcp/contrast/ADRServiceTest.java

```diff
@@ -539,6 +540,132 @@ void testGetAttacks_MultipleValidationErrors_CombinesErrors() throws Exception {
+        });
+
+        assertTrue(exception.getMessage().contains("Application ID cannot be null or empty"),
+            "Should have descriptive error message");
+    }
+
+    @Test
+    void testGetProtectDataByAppID_NullAppID() {
+        // When/Then
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
+            adrService.getProtectDataByAppID(null);
+        });
+
+        assertTrue(exception.getMessage().contains("Application ID cannot be null or empty"),
+            "Should have descriptive error message");
+    }
+
+    @Test
+    void testGetProtectDataByAppID_SDKFailure() throws Exception {
+        // Given - SDK throws exception
+        mockedSDKExtension = mockConstruction(SDKExtension.class, (mock, context) -> {
+            when(mock.getProtectConfig(eq(TEST_ORG_ID), anyString()))
+                .thenThrow(new RuntimeException("Failed to fetch protect config"));
+        });
+
+        // When/Then
+        Exception exception = assertThrows(Exception.class, () -> {
+            adrService.getProtectDataByAppID(TEST_APP_ID);
+        });
+
+        assertTrue(exception.getMessage().contains("Failed to fetch protect config") ||
+                   (exception.getCause() != null &&
+                    exception.getCause().getMessage().contains("Failed to fetch protect config")),
+            "Should propagate SDK exception");
+    }
+
+    @Test
+    void testGetProtectDataByAppID_NoProtectDataReturned() throws Exception {
+        // Given - SDK returns null (app exists but no protect config)
+        mockedSDKExtension = mockConstruction(SDKExtension.class, (mock, context) -> {
+            when(mock.getProtectConfig(eq(TEST_ORG_ID), eq(TEST_APP_ID)))
+                .thenReturn(null);
+        });
+
+        // When
+        com.contrast.labs.ai.mcp.contrast.sdkexstension.data.ProtectData result =
+            adrService.getProtectDataByAppID(TEST_APP_ID);
+
+        // Then
+        assertNull(result, "Should return null when no protect data available");
+    }
+
+    @Test
+    void testGetProtectDataByAppID_EmptyRulesList() throws Exception {
+        // Given - Protect enabled but no rules configured
+        com.contrast.labs.ai.mcp.contrast.sdkexstension.data.ProtectData mockProtectData =
+            new com.contrast.labs.ai.mcp.contrast.sdkexstension.data.ProtectData();
+        mockProtectData.setRules(new ArrayList<>());
+
+        mockedSDKExtension = mockConstruction(SDKExtension.class, (mock, context) -> {
+            when(mock.getProtectConfig(eq(TEST_ORG_ID), eq(TEST_APP_ID)))
+                .thenReturn(mockProtectData);
+        });
+
+        // When
+        com.contrast.labs.ai.mcp.contrast.sdkexstension.data.ProtectData result =
+            adrService.getProtectDataByAppID(TEST_APP_ID);
+
+        // Then
+        assertNotNull(result);
+        assertNotNull(result.getRules());
+        assertTrue(result.getRules().isEmpty(), "Should have empty rules list");
```

Comment:
not a big deal, but some devs like using https://assertj.github.io/doc/#assertj-core for assertions.  It reads a little more natural to a human.